build.gradle.kts

```diff
@@ -0,0 +1,375 @@
+                    sourceMap = true
+                    metaInfo = true
+                    moduleKind = "umd"
+                }
+            }
+        }
+
+        // Only enable when loaded in IDEA (we use a property for detection). In CLI that would produce an "expect" error.
+
+        if (System.getProperty("idea.version") != null) {
+            when {
+                Os.isFamily(Os.FAMILY_WINDOWS) -> run { mingwX64("nativeCommon"); mingwX64("nativePosix") }
+                Os.isFamily(Os.FAMILY_MAC) -> run { macosX64("nativeCommon"); macosX64("nativePosix") }
+                else -> run { linuxX64("nativeCommon"); linuxX64("nativePosix") }
+            }
+        }
+
+        sourceSets {
+            val jvmMain = this["jvmMain"]
+            val nativeCommonMain = create("nativeCommonMain")
+            val nativeCommonTest = create("nativeCommonTest")
+            val nativePosixMain = create("nativePosixMain")
+            if (hasAndroid) {
+                this.maybeCreate("androidMain").apply {
+                    dependsOn(jvmMain)
+                }
+                this.maybeCreate("androidTest").apply {
+                    dependsOn(jvmMain)
+                }
+            }
+            maybeCreate("mingwX64Main").apply {
+                dependsOn(nativeCommonMain)
+            }
+            maybeCreate("mingwX64Test").apply {
+                dependsOn(nativeCommonTest)
+            }
+
+            val iosTargets = listOf(this["iosX64Main"], this["iosArm32Main"], this["iosArm64Main"])
+
+            configure(iosTargets + listOf(this["macosX64Main"], this["linuxX64Main"])) {
+                dependsOn(nativeCommonMain)
+                dependsOn(nativePosixMain)
+            }
+
+            configure(iosTargets + listOf(this["macosX64Test"], this["linuxX64Test"])) {
+                dependsOn(nativeCommonTest)
+            }
+
+            val iosCommonMain = create("iosCommonMain")
+            val iosCommonTest = create("iosCommonTest")
+
+            configure(iosTargets) { dependsOn(iosCommonMain) }
+            configure(iosTargets) { dependsOn(iosCommonTest) }
+
+        }
+    }
+
+    dependencies {
+        commonMainImplementation("org.jetbrains.kotlin:kotlin-stdlib-common")
+        commonTestImplementation("org.jetbrains.kotlin:kotlin-test-annotations-common")
+        commonTestImplementation("org.jetbrains.kotlin:kotlin-test-common")
+
+        if (hasAndroid) {
+            add("androidMainImplementation", "org.jetbrains.kotlin:kotlin-stdlib")
+            add("androidTestImplementation", "org.jetbrains.kotlin:kotlin-test")
+            add("androidTestImplementation", "org.jetbrains.kotlin:kotlin-test-junit")
+        }
+
+        add("jsMainImplementation", "org.jetbrains.kotlin:kotlin-stdlib-js")
+        add("jsTestImplementation", "org.jetbrains.kotlin:kotlin-test-js")
+
+        add("jvmMainImplementation", "org.jetbrains.kotlin:kotlin-stdlib-jdk8")
```

Comment:
Is this the recommended way for adding dependencies? Is there something more dsl-like?